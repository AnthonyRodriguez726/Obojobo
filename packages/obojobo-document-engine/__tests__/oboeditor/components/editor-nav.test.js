import { mount } from 'enzyme'
import React from 'react'
import renderer from 'react-test-renderer'

import EditorNav from '../../../src/scripts/oboeditor/components/editor-nav'

import EditorUtil from '../../../src/scripts/oboeditor/util/editor-util'
import Common from '../../../src/scripts/common'

jest.mock('../../../src/scripts/oboeditor/util/editor-util')

describe('EditorNav', () => {
	beforeEach(() => {
		jest.clearAllMocks()

		Common.models.OboModel.models = {
			'5': {
				isFirst: () => true,
				isLast: () => false,
				getIndex: () => 0
			},
			'6': {
				isFirst: () => false,
				isLast: () => true,
				getIndex: () => 1
			}
		}
	})

	test('EditorNav component', () => {
		EditorUtil.getOrderedList.mockReturnValueOnce([])
		const props = {
			navState: {}
		}
		const component = renderer.create(<EditorNav {...props} />)
		const tree = component.toJSON()
		expect(tree).toMatchSnapshot()
	})

	test('EditorNav component with all list item types', () => {
		EditorUtil.getOrderedList.mockReturnValueOnce([
			{
				id: 4,
				type: 'heading',
				label: 'label4'
			},
			{
				id: 5,
				type: 'link',
				label: 'label5',
				flags: {
					assessment: false
				}
			},
			{
				id: 6,
				type: 'link',
				label: 'label6',
				flags: {
					assessment: true
				}
			},
			{
				id: 7,
				type: 'sublink',
				label: 'label7',
				flags: {
					assessment: false
				}
			}
		])

		const props = {
			navState: {
				open: false,
				locked: true,
				navTargetId: 56 // select this item
			}
		}
		const component = renderer.create(<EditorNav {...props} />)
		const tree = component.toJSON()
		expect(tree).toMatchSnapshot()
	})

	test('EditorNav component clicks Add Page button', () => {
		EditorUtil.getOrderedList.mockReturnValueOnce([]).mockReturnValueOnce([])
		jest.spyOn(window, 'prompt')
		window.prompt.mockReturnValueOnce(null)

		const props = { navState: {} }
		const component = mount(<EditorNav {...props} />)

		component
			.find('button')
			.at(0)
			.simulate('click')

		expect(EditorUtil.addPage).toHaveBeenCalled()
	})

	test('EditorNav component clicks Add Assessment button', () => {
		EditorUtil.getOrderedList.mockReturnValueOnce([]).mockReturnValueOnce([])
		jest.spyOn(window, 'prompt')
		window.prompt.mockReturnValueOnce(null)

		const props = { navState: {} }
		const component = mount(<EditorNav {...props} />)

		component
			.find('button')
			.at(1)
			.simulate('click')

		expect(EditorUtil.addAssessment).toHaveBeenCalled()
	})

	test('EditorNav component clicks Move Up button in item', () => {
		EditorUtil.getOrderedList
			.mockReturnValueOnce([
				{
					id: 6,
					type: 'link',
					label: 'label6',
					flags: {
						assessment: true
					}
				}
			])
			.mockReturnValueOnce([])

		const props = { navState: {} }
		const component = mount(<EditorNav {...props} />)

		component
			.find('li')
			.find('button')
			.at(0) // [Move Up, Edit Name, Delete]
			.simulate('click')

		expect(EditorUtil.movePage).toHaveBeenCalled()
	})

	test('EditorNav component clicks Move Down button in item', () => {
		EditorUtil.getOrderedList
			.mockReturnValueOnce([
				{
					id: 5,
					type: 'link',
					label: 'label5',
					flags: {
						assessment: true
					}
				}
			])
			.mockReturnValueOnce([])

		const props = { navState: {} }
		const component = mount(<EditorNav {...props} />)

		component
			.find('li')
			.find('button')
			.at(0) // [Move Down, Edit Name, Delete]
			.simulate('click')

		expect(EditorUtil.movePage).toHaveBeenCalled()
	})

	test('EditorNav component clicks Edit Name button in item', () => {
		EditorUtil.getOrderedList
			.mockReturnValueOnce([
				{
					id: 5,
					type: 'link',
					label: 'label5',
					flags: {
						assessment: true
					}
				}
			])
			.mockReturnValueOnce([])
		jest.spyOn(window, 'prompt')
		window.prompt.mockReturnValueOnce(null)

		const props = { navState: {} }
		const component = mount(<EditorNav {...props} />)

		component
			.find('li')
			.find('button')
			.at(1) // [Move Down, Edit Name, Delete]
			.simulate('click')

		expect(EditorUtil.renamePage).toHaveBeenCalled()
	})

	test('EditorNav component clicks Delete button in item', () => {
		EditorUtil.getOrderedList
			.mockReturnValueOnce([
				{
					id: 5,
					type: 'link',
					label: 'label5',
					flags: {
						assessment: true
					}
				}
			])
			.mockReturnValueOnce([])
		jest.spyOn(window, 'prompt')
		window.prompt.mockReturnValueOnce(null)

		const props = { navState: {} }
		const component = mount(<EditorNav {...props} />)

		component
			.find('li')
			.find('button')
			.at(2) // [Move Down, Edit Name, Delete]
			.simulate('click')

		expect(EditorUtil.deletePage).toHaveBeenCalled()
	})
})
