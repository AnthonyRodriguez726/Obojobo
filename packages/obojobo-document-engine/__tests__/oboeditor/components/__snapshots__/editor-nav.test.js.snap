// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditorNav EditorNav component 1`] = `
<div
  className="viewer--components--nav is-not-locked is-not-open is-enabled"
>
  <button
    className="toggle-button"
  >
    Toggle Navigation Menu
  </button>
  <ul />
  <button
    onClick={[Function]}
  >
    Add Page
  </button>
  <button
    onClick={[Function]}
  >
    Add Assessment
  </button>
</div>
`;

exports[`EditorNav EditorNav component with all list item types 1`] = `
<div
  className="viewer--components--nav is-locked is-not-open is-enabled"
>
  <button
    className="toggle-button"
  >
    Toggle Navigation Menu
  </button>
  <ul>
    <li
      className="heading is-not-selected"
    >
      <a>
        label4
      </a>
    </li>
    <li
      className="link is-not-selected is-not-assessment is-not-first-in-list is-not-last-in-list"
      onClick={[Function]}
    >
      <a>
        label5
      </a>
      <div
        className="dropdown"
      >
        <span
          className="drop-arrow"
        >
          ▼
        </span>
        <div
          className="drop-content"
        >
          <button
            onClick={[Function]}
          >
            Move Down
          </button>
          <button
            onClick={[Function]}
          >
            Edit Name
          </button>
          <button
            onClick={[Function]}
          >
            Delete
          </button>
          <button>
            Id: 5
          </button>
        </div>
      </div>
    </li>
    <li
      className="link is-not-selected is-assessment is-not-first-in-list is-last-in-list"
      onClick={[Function]}
    >
      <a>
        label6
      </a>
      <div
        className="dropdown"
      >
        <span
          className="drop-arrow"
        >
          ▼
        </span>
        <div
          className="drop-content"
        >
          <button
            onClick={[Function]}
          >
            Move Up
          </button>
          <button
            onClick={[Function]}
          >
            Edit Name
          </button>
          <button
            onClick={[Function]}
          >
            Delete
          </button>
          <button>
            Id: 6
          </button>
        </div>
      </div>
    </li>
  </ul>
  <button
    onClick={[Function]}
  >
    Add Page
  </button>
  <button
    onClick={[Function]}
  >
    Add Assessment
  </button>
</div>
`;
